### PR TITLE
US 171 Deploy Client

### PR DESCRIPTION
**Pull request by aneuhold**
## Overview

This PR implements a client deployment method, and converts our Travis CI continuous integration into Github Actions. It also fixes some build issues that were happening.

### Conversion to Github Actions

This was done because Travis CI was preventing us from making any more builds or tests due to an overage of minutes. There also wasn't a way for the team to see how many minutes we were using. Because of this, the CI has been moved over to Github Actions. All previous functionality has been successfully ported.

Whenever a workflow triggers, we see similar information to Travis CI, for example:

Build example

### Small change to `run.sh`

The dependency installation was moved into its own part of `run.sh`. This was because it was taking an excessive amount of log space and time to run through the dependencies multiple times in the same testing run. This way dependencies are only checked once, and then the other actions are performed. 

### Deployment of client

Right now, until we discuss with Steven the permissions aspect and configuring the repo to either deploy to github pages, or maybe set the repo up under an org so we all have admin privileges, the client cannot be deployed. A production build has been tested locally though in the meantime to make sure that it should work once it does get deployed. It seemed like we could discuss this and set this up at the review meeting. A Netlify account has been created for the team ahead of time just in case we use that. It is not connected to the repo in any way at the moment. 

## Includes:
- .github/workflows/client.yaml 
- .github/workflows/clientDeployment.yaml 
- .github/workflows/server.yaml 
- .travis.yml 
- README.md 
- client/.travis.yml 
- client/package-lock.json 
- client/package.json 
- client/public/index.html 
- client/src/components/Graph.vue 
- client/src/utils/Store.js 
- server/.flake8 
- server/.travis.yml 
- server/README.md 
- server/config.py 
- server/halomod_app/__init__.py 
- server/run.sh 

## Developer Checklist:
- [x] The code follows the Quality Policy file on Google Drive
- [x] My code has been developer-tested and includes unit tests
- [x] I have considered proper use of exceptions
- [x] I have eliminated IDE warnings

## Notes:
- autopep8 did some formatting changes to a few files.

## Refs:
- [US 89](https://the-halo-mod.atlassian.net/jira/software/projects/HM/boards/1?selectedIssue=HM-89)
- [Storyless Task 185](https://the-halo-mod.atlassian.net/jira/software/projects/HM/boards/1?selectedIssue=HM-185)

### CODE REVIEW

**Review by aschnapp — APPROVED** [review]
Good work!

File: server/run.sh [hunk 1]
```diff
@@ -1,25 +1,28 @@
+#!/bin/bash
+
 for arg in "$@"
 do 
   case $arg in
     "--ci" )
-      python3 -m pip install -r requirements.txt;;
-    * )
+      ;;
```
Comment (aschnapp) [inline server/run.sh:7]: is this unused right now? Why not just delete it?

**Review by aschnapp — CHANGES_REQUESTED** [review]
I've approved it, if this is used, just dismiss this review.

File: server/run.sh (hunk 1 above)
Comment (aneuhold) [reply server/run.sh:7]: Good call out. Wasn't sure if it was going to be needed yet, so left it in. Just forgot to take it out. Working on that now.

Comment (aneuhold) [reply server/run.sh]: Done :)

**Review by vperuzzi — APPROVED** [review]

**Pull request merged**